pkg/api/defaults.go

```diff
@@ -120,7 +120,9 @@ func (cs *ContainerService) setOrchestratorDefaults(isUpgrade, isScale bool) {
 		case NetworkPolicyCilium:
 			o.KubernetesConfig.NetworkPlugin = NetworkPluginCilium
 		case NetworkPolicyAntrea:
-			o.KubernetesConfig.NetworkPlugin = NetworkPluginAntrea
+			if o.KubernetesConfig.NetworkPlugin == "" {
+				o.KubernetesConfig.NetworkPlugin = NetworkPluginAntrea
```

Comment:
why don't we leave default to azure cni? as like other plugins( calico, cilium)

Reply:
We would like to support both Encap Mode and Policy Only Mode for Antrea. When no NetworkPlugin is selected, Antrea will take care of networking and policies. and when Azure CNI is selected as Network Plugin, Antrea will operate in Policy Mode

Reply:
It should be other way. When no network plugin is selected, the default network plugin should be azure cni. Antrea network plugin should be selected when only user specifies it.

Reply:
We discuss this with Jack sometime back.

Antrea : just as networkPlugin. Invalid option
Antrea : as PolicyPlugin, default has to be Antrea for NetworkPlugin as it doesnt support any CNI chain with other networkPlugin

With this checkin
Antrea: as PolicyPlugin, Azure CNI **can** be selected as Network Plugin

Reply:
I think @aanandr discussed with few people in your team long back about integrating antrea network policy with azure cni plugin and this was not part of the plan.

Reply:
Also, other plugins do follow same model.

Reply:
calico with kubenet was there even before azure cni integrated with AKS. So we cannot break the existing scenario. I wasn't aware of cilium changes.

Reply:
> I think @aanandr discussed with few people in your team long back about integrating antrea network policy with azure cni plugin and this was not part of the plan.

What you referring to?? Not sure if we are on same page now. Isnt it just networkPlugin selection or you referring to not supporting Encap mode at all??

Reply:
Plan is to support Antrea as networkPolicy plugin with Azure CNI in Azure managed service AKS. So, I am assuming user will select Azure CNI and Antrea CNI at the time of deploying the cluster. 
Wrt to defaults, we can also make Azure CNI as default in AKS-Engine when Antrea CNI is selected for Policy. But I dont see any harm in other way around too. Explicit selection of Azure CNI makes sense to me. 

Also,  "calico with kubenet was there even before azure cni integrated with AKS. So we cannot break the existing scenario. " I didnt get this? Which scenario is this checkin breaking?